src/components/pagination/docs.mdx

```diff
@@ -0,0 +1,101 @@
+---
+componentName: Pagination
+---
+
+## Overview
+
+The Pagination component allows users to navigate through a large dataset split into pages. These pages could be either links (different URLs) or content rendered within the same page/URL.
+
+It is not meant to be used as a navigation control for regular content or as a stepper, i.e. a guide, tutorial, setup or creation flow.
+
+### `Pagination`
+
+This is a container component that provides grid layout for the subcomponents and pagination react context.
+
+**Props:**
+
+- `totalItems` (required): Total number of items
+- `pageSize` (required): Number of items per page
+- `page` (optional): Current page index. This is 1-based, and is only used for the initial render. If not provided, the component will use the first page by default.
+- `onPageChange` (optional): Callback function to be called when a page is selected. It receives the page index as an argument.
+- `onPageSizeChange` (optional): Callback function to be called when a page size is selected. It receives the page size as an argument.
+
+### `Pagination.Info`
+
+Displays info like, _"90-100 of 103 items"_
+
+**Props:**
+
+- `showTotalItems`: Whether to show the total number of items. Default: `false``
+
+### `Pagination.Nav`
+
+Displays navigation buttons.
+
+**Props:**
+
+- `type`: "compact" | "numbered" | "truncated"
+  - "compact": Displays only previous and next buttons
+  - "numbered": Displays numbered buttons for each page
+  - "truncated": Displays numbered buttons for each page, but truncates the list if there are too many pages
+
+### `Pagination.SizeSelector`
+
+Displays a dropdown to select the page size.
+
+- `sizes`: Array of page size numbers
+
+## Basic
+
+<LiveComponent>
+	{`<Pagination
+	totalItems={103}
+	pageSize={10}
+	page={3}
+	onPageChange={(idx, pagesize) => console.log("Select page: %s; Pagesize: %s", idx, pagesize)}
+>
+	<Pagination.Info showTotalItems={true}/>
+	<Pagination.Nav type="truncated" />
+	<Pagination.SizeSelector sizes={[10, 30, 50]}/>
```

Comment:
Curious of the motivation here to use some children items to specify some configuration.  Is this the interface of the other HDS implementation? Also too, curious if there are any further pros/cons to this approach you can share -- I'm not super familiar with approaching managing the config this way (usually I'd just reach for props on the top level component).

Reply:
p.s. I definitely don't object to this setup at all, just genuinely curious!

Reply:
I think this — the _composition_ pattern — avoids the case of having a "black-box-ish" top-level `Pagination` component.

Pro: If you go to the Swingset - "With Mobile Styling" example, you'll see that you can very easily conditionally render various composed children components. (read: there's more flexibility/freedom)

Con: You could theoretically and potentially enable passing props to _either_ the top-level or composed-children components, depending on how you write the internals. (which raises the authoring complexity)

I haven't thought too hard or been dogmatic about, _"Oh props can ONLY be passed here or cannot be passed there"_

These my own moderate opinions, loosely held 😂 

Good example in the wild — https://cmdk.paco.me/